src/linked_list.py

```diff
@@ -0,0 +1,82 @@
+"""Creating a Linked List Data Structure.  CF 401 Python Week 2 assignment."""
+
+
+class Node(object):
+    """Create Node objects for use in a Linked List data structure."""
+    def __init__(self, value=None, nxt=None):
+        self.value = value
+        self.nxt = nxt
+
+
+class Linked_List(object):
+    """Create a Linked List Data Structure."""
+    def __init__(self, iterable=None):
+        """Intialize a Linked List Object."""
+        self.head = None
+        self._size = 0
+        if iterable and hasattr(iterable, "__iter__"):
+            for value in iterable:
+                self.push(value)
+        elif iterable:
+            raise TypeError
+
+    def push(self, value):
+        """Add a node."""
+        self.head = Node(value, self.head)
+        self._size += 1
+
+    def size(self):
+        """Return the length of the Linked_List."""
+        return self._size
+
+    def pop(self):
+        """Pop the first value off the head of LL and return it."""
+        if self.head is None:
+            raise IndexError("Cannot pop from an empty Linked List.")
+        val = self.head.value
+        self.head = self.head.nxt
+        self._size -= 1
+
+        return val
+
+    def search(self, query):
+        """Return the node that contains the value."""
+        match = [node for node in self._iterate_from(self.head) if node.value == query]
+        return match[0] if match else "That value is not in this linked list."
```

Comment:
Actually this is supposed to return `None` if no nodes in the list contain the value.